src/main/java/nl/rijksoverheid/mev/logging/LoggingContext.java

```diff
@@ -0,0 +1,87 @@
+package nl.rijksoverheid.mev.logging;
+
+import nl.rijksoverheid.mev.gezagsmodule.model.Burgerservicenummer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.util.*;
+
+public class LoggingContext {
+
+    private static final Logger logger = LoggerFactory.getLogger(LoggingContext.class);
+
+    private final Instant eventStart;
+    private final List<Long> plIds;
+    private final Map<String, GezagResultaat> gezagResultaten;
+
+    public LoggingContext() {
+        this.eventStart = Instant.now();
+        this.plIds = new ArrayList<>();
+        this.gezagResultaten = new HashMap<>();
+    }
+    
+    private static String mapGezagType(String gezagType) {
+        // temporary until GezagService returns typed responses
+        return switch (gezagType) {
+            case "OG1" -> "EenhoofdigOuderlijkGezag";
+            case "OG2" -> "TweehoofdigOuderlijkGezag";
+            case "GG" -> "GezamenlijkGezag";
+            case "V" -> "Voogdij";
+            case "G" -> "TijdelijkGeenGezag";
+            case "N" -> "GezagNietTeBepalen";
+            default -> { // should never happen
+                logger.error("Unknown gezag type: {}", gezagType);
+                yield "OnbekendGezagType";
+            }
+        };
+    }
+
+    public void addBurgerservicenummers(Collection<String> burgerservicenummers) {
+        burgerservicenummers.forEach(burgerservicenummer ->
+            gezagResultaten.put(burgerservicenummer, new GezagResultaat())
+        );
+    }
+
+    public void addPlIdIfBurgerservicenummerMatches(long plId, Burgerservicenummer burgerservicenummer) {
```

Comment:
smaak dingetje, kijk maar of je er wat met doet.

addPlId -> als burgerservicenummer matches
Maar de plId wordt altijd toegevoegd in iedergeval aan de plIds, en wordt gezet in het gezagresultaat als het matched.

Misschien gewoon `addPlId` van maken?

Reply:
Goed idee. Overgenomen.